src/TLuaInterpreter.cpp

```diff
@@ -13769,3 +13770,40 @@ int TLuaInterpreter::startPermSubstringTrigger(const QString & name, const QStri
 
 }
 
+int TLuaInterpreter::alert( lua_State * L )
+{
+    int luaAlertDuration;
+
+    int n = lua_gettop( L );
+    if( n == 0 )
+    {
+        // msec argument for QApplication::alert() is optional (defaults to 0)
+        luaAlertDuration = 0;
+    }
+    else
+    {
+        if( ! lua_isnumber( L, 1 ) )
+        {
+            lua_pushfstring( L, "alert: bad argument #1 type (alert duration in seconds as number expected, got %s!)",
+                            luaL_typename( L, 1 ));
+            lua_error( L );
+            return 1;
+        }
+        else
+        {
+            luaAlertDuration = lua_tointeger( L, 1 );
+
+            if ( luaAlertDuration < 0 )
+            {
+                lua_pushstring( L, "alert: duration, in seconds, is optional but if given must be zero or greater." );
+                lua_error( L );
+                return 1;
+            }
+        }
+    }
+
+    // QApplication::alert expects milliseconds, not seconds
+    QApplication::alert(mudlet::self(), luaAlertDuration * 1000);
```

Comment:
... finally convert a decimal number of seconds to the integer number wanted with:

    QApplication::alert(mudlet::self(), qRound(luaAlertDuration * 1000.0));

Now the code will handle an argument of `0.3` seconds just as well as `3`! 😀